docs/components/reference.mdx

```diff
@@ -0,0 +1,1180 @@
+        {
+            "type": 2,
+            "label": "Decline",
+            "style": 4,
+            "custom_id": "click_no"
+        }
+    ]
+  }
+]
+```
+
+![Example of an Action Row with three buttons](images/components/action-row.png)
+
+## Button
+
+A Button is an interactive component that can only be used in messages. It creates clickable elements that users can interact with, sending an [interaction](/docs/interactions/receiving-and-responding#interaction-object) to your app when clicked.
+
+Buttons must be placed inside an [Action Row](/docs/components/reference#action-row) or a [Section](/docs/components/reference#section)'s `accessory` field.
+
+###### Button Structure
+
+| Field     | Type                                                | Description                                                                                                         |
+|-----------|-----------------------------------------------------|---------------------------------------------------------------------------------------------------------------------|
+| type      | integer                                             | `2` for a button                                                                                                    |
+| id?       | integer                                             | Optional identifier for component                                                                                   |
+| style     | integer                                             | A [button style](/docs/components/reference#button-button-styles)                                                   |
+| label?    | string                                              | Text that appears on the button; max 80 characters                                                                  |
+| emoji?    | partial [emoji](/docs/resources/emoji#emoji-object) | `name`, `id`, and `animated`                                                                                        |
+| custom_id | string                                              | Developer-defined identifier for the button; max 100 characters                                                     |
+| sku_id?   | snowflake                                           | Identifier for a purchasable [SKU](/docs/resources/sku#sku-object), only available when using premium-style buttons |
+| url?      | string                                              | URL for link-style buttons                                                                                          |
+| disabled? | boolean                                             | Whether the button is disabled (defaults to `false`)                                                                |
+
+Buttons come in various styles to convey different types of actions. These styles also define what fields are valid for a button.
+
+- Non-link and Non-premium buttons **must** have a `custom_id`, and cannot have a `url` or a `sku_id`.
```

Comment:
```suggestion
- Non-link and non-premium buttons **must** have a `custom_id`, and cannot have a `url` or a `sku_id`.
```
Unsure but lowercase n because it's not the beginning of the sentence?